### PR TITLE
Dash extra menu items

### PR DESCRIPTION
**Pull request by ghsolomon**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by ghsolomon** [comment]
See Toolbox PR: https://github.com/xh/toolbox/pull/590

File: desktop/cmp/contextmenu/ContextMenu.js [hunk 1]
```diff
@@ -49,7 +48,9 @@ ContextMenu.propTypes = {
 function parseMenuItems(items) {
     items = items.map(item => {
         if (item === '-' || isValidElement(item)) return item;
-
+        if (item.displayFn) {
+            item = {...item, ...item.displayFn()};
```
Comment (lbwexler) [inline desktop/cmp/contextmenu/ContextMenu.js:52]: this displayFn is a little different than the one used by RecordAction, which actually takes the existing item, and returns a new config.  That setup can be useful when trying to write a general display function.  Not sure what the right thing to do here is.

File: desktop/cmp/dash/canvas/DashCanvasModel.js [hunk 2]
```diff
@@ -43,6 +43,8 @@ export class DashCanvasModel extends HoistModel {
     @bindable margin;
     /** @member {number[]} - [paddingX, paddingY] */
     @bindable containerPadding;
+    /** @member {ContextMenuItem[]|Object[]|string[]} */
+    @bindable extraMenuItems;
```
[the remark this answers is not in the document]
Comment (ghsolomon) [reply desktop/cmp/dash/canvas/DashCanvasModel.js:47]: Thanks `bindable.ref` is actually what I meant to do here.  Any reason not to?

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,9 @@
   a remount/rerender of the panel's contents.
 * FilterChooser field suggestions now search within multi-word field names.
 * Autosize performance has been improved for very large grids.
+* `DashCanvas.extraMenuItems` and `DashContainer.extraMenuItems` are now observable properties
+* `ContextMenuItem.hidden` and `MenuItem.hidden` now additionally accept callback functions to
```
[the remark this answers is not in the document]
Comment (ghsolomon) [reply CHANGELOG.md]: > Greg -- were you reading my mind?

Wish I could say I was!  @haynesjm42 pointed me to look at `RecordAction.displayFn`

**Pull request closed without merging**